### ISSUE TITLE
.lnk files are wrong if you install overwritting installdir

### ISSUE THREAD

**Issue by jmlucjav** [mintty/wsltty#119]
I install wsltty overwritting both installdir and configdir.

Most stuff works fine (I see some error messages but not sure if they actually signal a real issue or no), but the lnk files that are droped at c:\Users\...\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\WSLtty\ have the paths hardcoded to "%LOCALAPPDATA%\wsltty so they don't point to the right installation dir.

**Comment by mintty** [comment]
The information is implicitly (by environment variable) passed from install.bat to config-distros.sh and would need to be considered there when setting `target=`. It's then further passed to mkshortcut.vbs which should then work unchanged.

**Comment by jmlucjav** [comment]
I am not sure I follow sorry, my cmd/sh/vbs is quite weak/inexistent. 

About  config-distros.sh...when running 'install myinstalldir myconfigdir' I saw an error from  config-distros.sh, something about basename. I did not keep the error sorry (but I could reproduce if needed). What I mean, not sure if there are more errors down the line that would still make some things not work. I can test things if you need me to test any change.

thanks

**Comment by mintty** [comment]
> my cmd/sh/vbs is quite weak/inexistent.

Why do you actually want to install in a non-default location? This installation option is marked "for experts only".

**Comment by jmlucjav** [comment]
I use several computers, and I want to keep my HOME under dropbox so changes I apply in one automatically apply to others. When I said my cmd/sh skills are weak I mean I am not an expert (vbs I never used before), but I can get by.

**Comment by mintty** [comment]
Please test.

**Comment by jmlucjav** [comment]
hi, thanks for taking a stab at this...

I tested with the lastest commit, and this happens when I launch the installer (%mydir% is where I want wsltty to install, points to c:\Users\xxx\Dropbox\portable):

```
C:\Users\xxx\Dropbox\scripts\install\wsl\wsltty-master-2018-09-15>        install.bat %mydir%\wsltty %mydir%\wsltty
Could Not Find c:\Users\xxx\Dropbox\portable\wsltty\*.bat
Could Not Find c:\Users\xxx\Dropbox\portable\wsltty\*.lnk
        1 file(s) copied.
        1 file(s) copied.
        1 file(s) copied.
        1 file(s) copied.
        1 file(s) copied.
        1 file(s) copied.
The system cannot find the file specified.
The system cannot find the file specified.
        1 file(s) copied.
        1 file(s) copied.
        1 file(s) copied.
The system cannot find the file specified.
The system cannot find the file specified.
The system cannot find the file specified.
The system cannot find the file specified.
The system cannot find the file specified.
The system cannot find the file specified.
The system cannot find the file specified.
The system cannot find the file specified.
The system cannot find the file specified.
The system cannot find the file specified.
The system cannot find the file specified.
        1 file(s) copied.
Could Not Find C:\Users\xxx\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\WSLtty\*.lnk
        1 file(s) copied.
        1 file(s) copied.
        1 file(s) copied.
        1 file(s) copied.
        1 file(s) copied.
The system cannot find the file specified.
The system cannot find the file specified.
The system cannot find the file specified.
'"c:\Users\xxx\Dropbox\portable\wsltty\bin\zoo"' is not recognized as an internal or external command,
operable program or batch file.
'"c:\Users\xxx\Dropbox\portable\wsltty\bin\zoo"' is not recognized as an internal or external command,
operable program or batch file.
The system cannot find the path specified.
The system cannot find the path specified.
'bin\dash.exe' is not recognized as an internal or external command,
operable program or batch file.
c:\Users\xxx\Dropbox\portable\wsltty>
```

I think the problem is still there: c:\Users\xxx\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\WSLtty\add default to context menu.lnk has the following:

```
Target:	C:\Users\xxx\AppData\Local\wsltty\bin\dash.exe
Arguments:	"%LOCALAPPDATA%\wsltty\config-distros.sh" -contextmenu-default
Working folder:	%LOCALAPPDATA%\wsltty
```

still using hardcoded paths?

Also, something else went wrong, not sure if you can see it from the error logs, but as a result c:\Users\xxx\Dropbox\portable\wsltty\bin\ is empty, for example. I don't know if this is related to the original issue or not...

**Comment by mintty** [comment]
If your %installdir%\bin directory is empty, the config-distros.sh script could not even be run, so your resulting shortcuts are still your older ones.
Did you run the install.bat script from a folder which contains all the files to be installed, like dash.exe and zoo.exe, like the .exe installer does? Then it should work...
To test just the config script within cygwin, you could also run it like `installdir='C:\...' ./config-distros.sh`.

**Comment by jmlucjav** [comment]
ok, I think I missed somethign here....if there is no up to date wsltty...-install.exe available, and I want to use the latest commit, how should I proceed to install in a custom location?

What I was doing was:

1. download as zip
2.         install.bat %mycusctomdir%\wsltty %mycusctomdir%\wsltty

is this way ok? Note that I don't have cygwin installed.

thanks

**Comment by mintty** [comment]
You will basically need to clone the installation process and just substitute the install.bat invocation.
You cannot install from the repository, you need a build. You can either make that yourself, using cygwin, or use the provided installer to install to the default location, take the files from there and repeat installation as you like. As quote before, for experts only.

**Comment by jmlucjav** [comment]
I don't use cygwin anymore, since I moved to wsl, so, I think I'll wait till you release 1.9.1 and then I'll try the installer. Will report back. Thanks

**Comment by mintty** [comment]
Released wsltty 1.9.2.

**Comment by jmlucjav** [comment]
Tested with 1.9.2 installer, the files in c:\...\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\WSLtty\ still contain bad paths, for example:

 Information about shortcut file C:\Users\xxx\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\WSLtty\WSL Terminal.lnk:

>  Target:	C:\Users\xxx\AppData\Local\wsltty\bin\mintty.exe
> Arguments:	--WSL= --configdir="%APPDATA%\wsltty" -~
> Working folder:	%USERPROFILE%
> Shortcut key:	None
> Window:	Normal
> Comments:	
> Path to icon:	%LOCALAPPDATA%/wsltty/wsl.ico
> 	icon number: 0

This is the installer invokation in case it helps:

`
C:\wsltty1.9.2>        install.bat %DPORTABLE%\wsltty %DPORTABLE%\wsltty                                   
Could Not Find c:\Users\xxx\Dropbox\portable\wsltty\*.bat                                                                                        
Could Not Find c:\Users\xxx\Dropbox\portable\wsltty\*.lnk                                                                                        
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
Could Not Find C:\Users\xxx\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\WSLtty\*.lnk                                                   
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
The system cannot find the file specified.                                                                                                      
Zoo:  WARNING:  Archive header failed consistency check.                                                                                        
Zoo:  de.po          -- extracted                                                                                                               
Zoo:  en.po          -- extracted                                                                                                               
Zoo:  en_US.po       -- extracted                                                                                                               
Zoo:  es.po          -- extracted                                                                                                               
Zoo:  fr.po          -- extracted                                                                                                               
Zoo:  ja.po          -- extracted                                                                                                               
Zoo:  pt_BR.po       -- extracted                                                                                                               
Zoo:  ru.po          -- extracted                                                                                                               
Zoo:  zh_CN.po       -- extracted                                                                                                               
Zoo:  WARNING:  Archive header failed consistency check.                                                                                        
Zoo:  dracula        -- extracted                                                                                                               
Zoo:  flat-ui        -- extracted                                                                                                               
Zoo:  html           -- extracted                                                                                                               
Zoo:  mintty         -- extracted                                                                                                               
Zoo:  nord           -- extracted                                                                                                               
Zoo:  rosipov        -- extracted                                                                                                               
Zoo:  vga            -- extracted                                                                                                               
Zoo:  xterm          -- extracted                                                                                                               
The system cannot find the path specified.                                                                                                      
The system cannot find the path specified.                                                                                                      
distro 'Legacy'                                                                                                                                 
- name 'Bash on Windows'                                                                                                                        
- guid {12345678-1234-5678-0123-456789abcdef}                                                                                                   
- (launcher C:\WINDOWS/System32/bash.exe)                                                                                                       
- icon C:\Users\xxx\AppData\Local/lxss/bash.ico                                                                                                  
- root C:\Users\xxx\AppData\Local\lxss                                                                                                           
Creating Bash on Windows Terminal %.lnk                                                                                                         
Created Bash on Windows Terminal %.lnk                                                                                                          
                                                                                                                                                
        1 file(s) copied.                                                                                                                       
Creating "Bash on Windows".bat                                                                                                                  
The file cannot be copied onto itself.                                                                                                          
        0 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
Creating Bash on Windows Terminal.lnk                                                                                                           
Created Bash on Windows Terminal.lnk                                                                                                            
                                                                                                                                                
        1 file(s) copied.                                                                                                                       
Creating "Bash on Windows~".bat                                                                                                                 
The file cannot be copied onto itself.                                                                                                          
        0 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
distro 'Legacy'                                                                                                                                 
- name 'Bash on Windows'                                                                                                                        
- guid DefaultUsername                                                                                                                          
- (launcher C:\WINDOWS/System32/bash.exe)                                                                                                       
- icon C:\Users\xxx\AppData\Local/lxss/bash.ico                                                                                                  
- root C:\Users\xxx\AppData\Local\lxss                                                                                                           
distro 'Legacy'                                                                                                                                 
- name 'Bash on Windows'                                                                                                                        
- guid DefaultUid                                                                                                                               
- (launcher C:\WINDOWS/System32/bash.exe)                                                                                                       
- icon C:\Users\xxx\AppData\Local/lxss/bash.ico                                                                                                  
- root C:\Users\xxx\AppData\Local\lxss                                                                                                           
distro 'Legacy'                                                                                                                                 
- name 'Bash on Windows'                                                                                                                        
- guid DefaultGid                                                                                                                               
- (launcher C:\WINDOWS/System32/bash.exe)                                                                                                       
- icon C:\Users\xxx\AppData\Local/lxss/bash.ico                                                                                                  
- root C:\Users\xxx\AppData\Local\lxss                                                                                                           
distro 'Legacy'                                                                                                                                 
- name 'Bash on Windows'                                                                                                                        
- guid DefaultDistribution                                                                                                                      
- (launcher C:\WINDOWS/System32/bash.exe)                                                                                                       
- icon C:\Users\xxx\AppData\Local/lxss/bash.ico                                                                                                  
- root C:\Users\xxx\AppData\Local\lxss                                                                                                           
distro ''                                                                                                                                       
- name 'WSL'                                                                                                                                    
- guid                                                                                                                                          
- (launcher C:\WINDOWS/System32/bash.exe)                                                                                                       
- icon c:\Users\xxx\Dropbox\portable\wsltty\wsl.ico                                                                                              
- root C:\Users\xxx\AppData\Local\lxss                                                                                                           
Creating WSL Terminal %.lnk                                                                                                                     
Created WSL Terminal %.lnk                                                                                                                      
                                                                                                                                                
        1 file(s) copied.                                                                                                                       
Creating WSL.bat                                                                                                                                
The file cannot be copied onto itself.                                                                                                          
        0 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
Creating WSL Terminal.lnk                                                                                                                       
Created WSL Terminal.lnk                                                                                                                        
                                                                                                                                                
        1 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
Creating WSL~.bat                                                                                                                               
The file cannot be copied onto itself.                                                                                                          
        0 file(s) copied.                                                                                                                       
        1 file(s) copied.                                                                                                                       
                                                                                                            
`

**Comment by mintty** [comment]
> The file cannot be copied onto itself.
> 0 file(s) copied.

Maybe this is the time I should add logging capability to the install procedure.

Anyway, please check whether the .lnk files in question are really new ones, and whether the originally created ones (in %DPORTABLE%\wsltty) are the same.
Also, it may be a problem that you use the same folder both as installation and configuration directory:
> install.bat %DPORTABLE%\wsltty %DPORTABLE%\wsltty

Please try to use separate folders, as intended.

**Comment by jmlucjav** [comment]
> Anyway, please check whether the .lnk files in question are really new ones, and whether the originally created ones (in %DPORTABLE%\wsltty) are the same.
yes, positive, they are new

> Also, it may be a problem that you use the same folder both as installation and configuration directory:
> 
> > install.bat %DPORTABLE%\wsltty %DPORTABLE%\wsltty
> 
> Please try to use separate folders, as intended.

tested using different folders, and it does not make a difference. 

What I noticed is that SOME files have the paths (or most of them) ok:

Bash on Windows Terminal %.lnk: has all paths but 'path to icon' ok
WSL Terminal %.lnk: has all paths ok

The rest of the lnk have wrong paths.

Thanks!

**Comment by mintty** [comment]
Trying to reproduce as closely as possible by your description, I cannot reproduce the problem.
Installed with correct paths and icons.
I had asked you to check the timestamps of the shortcut files: please!
Maybe you previously installed as another user and are having access right problems now to overwrite old shortcut version?

**Comment by jmlucjav** [comment]
I did reply but my replies were merged with your quotes and maybe you didn't see them. Properly formated now:


> Anyway, please check whether the .lnk files in question are really new ones,
yes, positive, they are new.

>and whether the originally created ones (in %DPORTABLE%\wsltty) are the same.
they are NOT the same, at least not all of them. The ones in %DPORTABLE%\wsltty have more paths correctly pointing to my dirs (but not all files, and not all paths)

>Also, it may be a problem that you use the same folder both as installation and configuration directory:
>Please try to use separate folders, as intended.

tested using different folders, and it does not make a difference.

**Comment by mintty** [comment]
Can you delete the complete installation/configuration folder before new installation, please?

**Comment by jmlucjav** [comment]
I always delete it, they do not exist when the isntaller is run with 

install.bat %DPORTABLE%\wsltty %DPORTABLE%\wsltty

the installation creates them.

**Comment by mintty** [comment]
Is this still an issue?

**Comment by jmlucjav** [comment]
I have not reinstalled in some time, if you want you can close and I'll reopen if necessary when I test again.

**Comment by mintty** [comment]
Released 3.0.2.
